client/src/connection_cache.rs

```diff
@@ -257,9 +259,12 @@ struct GetConnectionResult {
     eviction_timing_ms: u64,
 }
 
-fn get_or_add_connection(addr: &SocketAddr) -> GetConnectionResult {
+fn get_or_add_connection(
```

Comment:
Yeah for sure! I wanted to minimize the changes, but it definitely makes more sense to move this inside of `ConnectionCache`.  So that I understand, you're suggesting it to be:

```
pub struct ConnectionCache {
    map: RwLock<IndexMap<SocketAddr, Arc<Connection>>>,
... the rest ...
}
```

Correct?

Reply:
Yup, and use_quic would probably need to be made atomic too.